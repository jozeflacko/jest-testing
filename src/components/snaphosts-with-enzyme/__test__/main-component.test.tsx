import * as Enzyme from 'enzyme';
import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import Main from '../main-component';

Enzyme.configure({ adapter: new Adapter() });

describe('<Main />', () => {
    const component = Enzyme.shallow(<Main />);
    
    it('renders correctly  component', () => {       
        expect(component).toMatchSnapshot();
    });

    it('child component is rendered as shallow',()=>{
        expect(component.find('Child').length).toBe(1);
    });

    console.log(component.find('Child').get(0));

});
