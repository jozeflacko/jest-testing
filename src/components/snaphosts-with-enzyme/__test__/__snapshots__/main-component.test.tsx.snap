// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders correctly  component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="hello"
        >
          
        </div>,
        <button
          onClick={[Function]}
        >
          Click
        </button>,
        <Child
          attrMsg="Ahoj"
          buttonLabel="Hello Jozef"
          onClicked={[Function]}
        />,
      ],
      "className": "main-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "hello",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Click",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "attrMsg": "Ahoj",
          "buttonLabel": "Hello Jozef",
          "onClicked": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="hello"
          >
            
          </div>,
          <button
            onClick={[Function]}
          >
            Click
          </button>,
          <Child
            attrMsg="Ahoj"
            buttonLabel="Hello Jozef"
            onClicked={[Function]}
          />,
        ],
        "className": "main-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "hello",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Click",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "attrMsg": "Ahoj",
            "buttonLabel": "Hello Jozef",
            "onClicked": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
